scylla-cdc/src/e2e_tests.rs

```diff
@@ -290,48 +290,38 @@ mod tests {
+            let mut is_ok = result.lock().await.iter_mut().fold(true, |acc, (pk, actual_operations)| acc && {
+                let mut expected_operations = match self.performed_operations.remove(pk) {
                     Some(ops) => ops,
                     None => {
                         eprintln!("Unexpected primary key {:?}", pk);
-                        is_ok = false;
-                        continue;
+                        return false;
                     }
                 };
                 let mut i = 0;
 
-                while let Some(next_expected) = expected_operations.pop_front() {
-                    i += 1;
-                    let next_actual = match actual_operations.pop_front() {
-                        Some(op) => op,
-                        None => {
-                            eprintln!("Too little read operations for primary key {:?}. Missing operations count: {}", pk, expected_operations.len() + 1);
-                            is_ok = false;
-                            break;
-                        }
-                    };
-
-                    if next_expected != next_actual {
-                        eprintln!("Operation no. {} not matching for primary key {:?}.", i, pk);
-                        eprintln!("\tExpected: {:?}, actual: {:?}", next_expected, next_actual);
-                        is_ok = false;
-                        break;
+                loop {
+                    match (expected_operations.pop_front(), actual_operations.pop_front()) {
+                        (Some(next_expected), Some(next_actual)) => {
+                            i += 1;
+                            if next_expected == next_actual {
+                                continue;
+                            }
+                            eprintln!("Operation no. {} not matching for primary key {:?}.", i, pk);
+                            eprintln!("\tExpected: {:?}, actual: {:?}", next_expected, next_actual);
+                        },
+                        (None, None) => return true,
+                        (None, _) => eprintln!("Too many read operations for primary key {:?}. Operations left: {}", pk, actual_operations.len() + 1),
+                        (_, None) => eprintln!("Too little read operations for primary key {:?}. Missing operations count: {}", pk, expected_operations.len() + 1),
                     }
+                    return false;
                 }
+            });
 
-                if is_ok && expected_operations.len() != actual_operations.len() {
-                    eprintln!(
-                        "Too many read operations for primary key {:?}. Operations left: {}",
-                        pk,
-                        actual_operations.len()
-                    );
-                    is_ok = false;
-                }
+            for pk in self.performed_operations.keys() {
+                eprintln!("Unexpected primary key {:?}", pk);
```

Comment:
The message is wrong - we did expect `pk` to be seen by the reader but it wasn't (IIUC) - also, it is exactly the same as the message on line `299/297` which will only make debugging harder when it pops up because it won't be possible to tell which line actually printed it. Please improve it.